pkg/util/initsystem/initsystem_windows.go

```diff
@@ -0,0 +1,172 @@
+	if err != nil {
+		return fmt.Errorf("could not start service %s: %v", service, err)
+	}
+	return nil
+}
+
+func (sysd WindowsInitSystem) ServiceRestart(service string) error {
+	if err := sysd.ServiceStop(service); err != nil {
+		return fmt.Errorf("couldn't stop service %s: %v", service, err)
+	}
+	if err := sysd.ServiceStart(service); err != nil {
+		return fmt.Errorf("couldn't start service %s: %v", service, err)
+	}
+	return nil
+}
+
+func (sysd WindowsInitSystem) ServiceStop(service string) error {
+	m, err := mgr.Connect()
+	if err != nil {
+		return err
+	}
+	defer m.Disconnect()
+
+	s, err := m.OpenService(service)
+	if err != nil {
+		return fmt.Errorf("could not access service %s: %v", service, err)
+	}
+	defer s.Close()
+
+	// Check if service is already stopped
+	status, err := s.Query()
+	if err != nil {
+		return fmt.Errorf("could not query service %s: %v", service, err)
+	}
+	if status.State == svc.Stopped {
+		return nil
+	}
+
+	// Stop the service and check the result
+	status, err = s.Control(svc.Stop)
+	if err != nil {
+		return fmt.Errorf("could not stop service %s: %v", service, err)
+	}
+	timeout := time.Now().Add(10 * time.Second)
+	for status.State != svc.Stopped {
+		if timeout.Before(time.Now()) {
+			return fmt.Errorf("timeout waiting for %s service to go to state=%d", service, svc.Stopped)
+		}
+		time.Sleep(300 * time.Millisecond)
+		status, err = s.Query()
+		if err != nil {
+			return fmt.Errorf("could not retrieve %s service status: %v", service, err)
+		}
+	}
+	return nil
+}
+
+func (sysd WindowsInitSystem) ServiceExists(service string) bool {
+	m, err := mgr.Connect()
+	if err != nil {
+		return false
+	}
+	defer m.Disconnect()
+	s, err := m.OpenService(service)
+	if err != nil {
+		return false
+	}
+	defer s.Close()
+
+	return true
+}
+
+func (sysd WindowsInitSystem) ServiceIsEnabled(service string) bool {
+	m, err := mgr.Connect()
+	if err != nil {
+		return false
+	}
+	defer m.Disconnect()
+
+	s, err := m.OpenService(service)
+	if err != nil {
+		return false
+	}
+	defer s.Close()
+
+	c, err := s.Config()
+	if err != nil {
+		return false
+	}
+
+	return c.StartType == mgr.StartAutomatic
+}
+
+func (sysd WindowsInitSystem) ServiceIsActive(service string) bool {
+	m, err := mgr.Connect()
+	if err != nil {
+		return false
+	}
+	defer m.Disconnect()
+	s, err := m.OpenService(service)
+	if err != nil {
+		return false
+	}
+	defer s.Close()
+
+	status, err := s.Query()
+	if err != nil {
+		return false
+	}
+	return status.State == svc.Running
+}
+
+// GetInitSystem returns an InitSystem for the current system, or nil
+// if we cannot detect a supported init system for pre-flight checks.
+// This indicates we will skip init system checks, not an error.
+func GetInitSystem() (InitSystem, error) {
+	m, err := mgr.Connect()
+	if err == nil {
+		defer m.Disconnect()
+		return &WindowsInitSystem{}, nil
+	}
+	return nil, fmt.Errorf("no supported init system detected, skipping checking for services")
```

Comment:
i'd suggest this modification:
```
	m, err := mgr.Connect()
	if err != nil {
		return nil, fmt.Errorf("no supported init system detected: %v", err)
	}
	defer m.Disconnect()
	return &WindowsInitSystem{}, nil
```
this also includes the `err` from `Connect()`, which can be useful for debug.

`skipping checking for services` should be on the CLI side.